libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartSyncPlugin/SFSmartSyncPlugin.m

```diff
@@ -199,15 +185,22 @@ - (void) reSync:(CDVInvokedUrlCommand *)command
 
 - (void) cleanResyncGhosts:(CDVInvokedUrlCommand *)command
 {
-    [self runCommand:^(NSDictionary* argsDict) {
-        NSNumber* syncId = (NSNumber*) [argsDict nonNullObjectForKey:kSyncIdArg];
-        [SFSDKHybridLogger d:[self class] format:@"cleanResyncGhosts with sync id: %@", syncId];
-        __weak typeof(self) weakSelf = self;
-        [[self getSyncManagerInst:argsDict] cleanResyncGhosts:syncId completionStatusBlock:^(SFSyncStateStatus syncStatus) {
-            [weakSelf handleGhostSyncUpdate:syncStatus];
-        }];
-        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
-    } command:command];
+    NSDictionary *argsDict = [self getArgument:command.arguments atIndex:0];
+    NSNumber* syncId = (NSNumber*) [argsDict nonNullObjectForKey:kSyncIdArg];
+    [SFSDKHybridLogger d:[self class] format:@"cleanResyncGhosts with sync id: %@", syncId];
+    id <CDVCommandDelegate> commandDelegate = self.commandDelegate;
+    NSString* callbackId = command.callbackId;
+    [[self getSyncManagerInst:argsDict] cleanResyncGhosts:syncId completionStatusBlock:^(SFSyncStateStatus syncStatus) {
+        CDVPluginResult* pluginResult;
+        if (syncStatus == SFSyncStateStatusDone) {
+            [SFSDKHybridLogger d:[self class] format:@"cleanResyncGhosts completed successfully"];
```

Comment:
Do we need the `weak`/`strong` dance on `self` inside this block to avoid a retain cycle?